### PR TITLE
PIConGPU: add fields and particles to header check

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add `fields` and `particle` header check
- fix missing includes and resolve cyclic includes

ci: picongpu

I tried to move as less as possible code but sometimes it was required to resolve cyclic includes.
I also removed two unused files. They were not included somewhere but broke the header test.

This PR is reducing the compile memory footprint to 78 GiB for KHI with nvcc 12.3

### CODE REVIEW

File: share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/atomicPhysics.param [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 #include "picongpu/particles/atomicPhysics/ExponentialApproximationProbability.hpp"
 #include "picongpu/particles/atomicPhysics/LinearApproximationProbability.hpp"
-#include "picongpu/particles/atomicPhysics/atomicData/AtomicData.hpp"
+#include "picongpu/particles/atomicPhysics/atomicData/AtomicData.def"
```
Comment (psychocoderHPC) [inline share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/atomicPhysics.param]: @BrianMarre this is something what we need to change in #5129, depending which PR is merged first.

File: include/picongpu/fields/absorber/Thickness.hpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/defines.hpp"
+
+#include <pmacc/traits/GetStringProperties.hpp>
```
Comment (ikbuibui) [inline include/picongpu/fields/absorber/Thickness.hpp]:
```suggestion
```
Looks unused

File: include/picongpu/particles/atomicPhysics/atomicData/AtomicData.def [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+     *
+     * Each dataBuffer will create on demand a host- or device-side dataBox objects which in turn gives access to the
+     *  data.
+     *
+     * Assumptions about input data are described in CheckTransitionTuple.hpp, ordering requirements of transitions in
+     *  CompareTransitionTuple.hpp and all other requirements in the checkChargeStateList(), checkAtomicStateList() and
+     *  checkTransitionsForEnergyInversion() methods.
+     *
+     * @todo add photonic channels, Brian Marre, 2022
+     */
```
Comment (ikbuibui) [inline include/picongpu/particles/atomicPhysics/atomicData/AtomicData.def:109]: This documentation is duplicated

File: include/picongpu/particles/atomicPhysics/stateRepresentation/DebugHelperConfigNumber.hpp [hunk 4]
```diff
@@ -1,152 +0,0 @@
-/* Copyright 2022-2023 Brian Marre
```
Comment (ikbuibui) [inline include/picongpu/particles/atomicPhysics/stateRepresentation/DebugHelperConfigNumber.hpp:1]: Just for me to know, why was this file removed?

**Review by ikbuibui — DISMISSED** [review]
"reducing the compile memory footprint to 78 GiB" :smiling_face_with_tear: 
Small comments, not blocking, can be changed in a follow up PR if appropriate
@BrianMarre can check if the removed files are really not required
@psychocoderHPC PR needs to be rebased

File: include/picongpu/particles/atomicPhysics/atomicData/AtomicData.def (hunk 3 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/atomicPhysics/atomicData/AtomicData.def:109]: well catched

File: include/picongpu/particles/atomicPhysics/kernel/CalculateStepLength.kernel [hunk 5]
```diff
@@ -22,8 +22,8 @@
 #pragma once
 
 #include "picongpu/defines.hpp"
-// need timeStepAlpha factor from atomicPhysics.param
-// need access to sim.pic.getDt() from simulation.unitless
+#include "picongpu/particles/atomicPhysics/localHelperFields/RejectionProbabilityCache.hpp"
```
Comment (BrianMarre) [inline include/picongpu/particles/atomicPhysics/kernel/CalculateStepLength.kernel:25]: This include is not required, the `CalculateStepLengthKernel` does not have any interaction with the `RejectionProbability` Cache, only the `RateCache`.

**Review by BrianMarre — APPROVED** [review]
The atomic physics look all okay.
As discussed offline I will re-add the removed functionality in a separate PRs.

**Review by ikbuibui — APPROVED** [review]
